lib/dry/cli.rb

```diff
@@ -145,29 +150,29 @@ def parse(command, arguments, names)
     # @since 0.6.0
     # @api private
     def build_command(command)
-      command.is_a?(Class) ? command.new : command
+      command.is_a?(Class) ? command.new(stderr: stderr, stdin: stdin, stdout: stdout) : command
     end
 
     # @since 0.6.0
     # @api private
     def help(command, prog_name)
-      out.puts Banner.call(command, prog_name)
+      stdout.puts Banner.call(command, prog_name)
       exit(0) # Successful exit
     end
 
     # @since 0.6.0
     # @api private
     def error(result)
-      err.puts(result.error)
+      stderr.puts(result.error)
       exit(1)
     end
 
     # @since 1.1.1
     def spell_checker(result, arguments)
       spell_checker = SpellChecker.call(result, arguments)
-      err.puts spell_checker if spell_checker
+      stderr.puts spell_checker if spell_checker
       puts
```

Comment:
@timriley is this a bug? Should this be `stderr.puts` and if so would you like me to fix it in this PR?